### PR TITLE
Feat/user self list rides

### PR DESCRIPTION
**Pull request by Scttpr**

### CODE REVIEW

**Comment by Scttpr** [comment]
[Historique de courses](https://trello.com/c/pf71hiNl/209-historique-de-courses)

File: middlewares/query-helper.mjs [hunk 1]
```diff
@@ -13,3 +15,85 @@ export const filtersToObject = (...filtersToParse) => async (ctx, next) => {
...
+  let alreadySet = false;
+
+  const queryFilter = filters.reduce((acc, current) => {
+    let newAcc;
+    const key = Object.keys(current).join('');
+
+    switch (key) {
+      case 'userId':
+        newAcc = {
+          $and: [
```
Comment (Nainterceptor) [inline middlewares/query-helper.mjs]: anti-pattern here

File: middlewares/query-helper.mjs [hunk 2]
```diff
@@ -13,3 +15,85 @@ export const filtersToObject = (...filtersToParse) => async (ctx, next) => {
...
+        if (startAndEnd.length < 2) {
+          ctx.throw_and_log(404, 'Start and end not found');
+        }
+        if (alreadySet) {
+          return acc;
+        }
+
+        alreadySet = true;
+        newAcc = {
+          $and: [
```
Comment (Nainterceptor) [inline middlewares/query-helper.mjs]: antipattern here too

File: middlewares/query-helper.mjs [hunk 3]
```diff
@@ -13,3 +15,85 @@ export const filtersToObject = (...filtersToParse) => async (ctx, next) => {
...
+              ),
+            },
+          ],
+        };
+        break;
+      }
+      case 'current': {
+        let queryType = '$nor';
+        if (current.current === 'false') {
+          queryType = '$and';
```
Comment (Nainterceptor) [inline middlewares/query-helper.mjs]: Anti-pattern here too

File: middlewares/query-helper.mjs [hunk 4]
```diff
@@ -13,3 +15,85 @@ export const filtersToObject = (...filtersToParse) => async (ctx, next) => {
...
+              ],
+            },
+          ],
+        };
+        break;
+      }
+      default:
+        newAcc = {
+          $and: [
+            ...acc.$and,
```
Comment (Nainterceptor) [inline middlewares/query-helper.mjs]: Anti-pattern

File: routes/rides.mjs [hunk 5]
```diff
@@ -191,6 +191,9 @@ const router = generateCRUD(Ride, {
       );
     },
   },
+  delete: {
+    right: [CAN_DELETE_SELF_RIDE],
```
Comment (Nainterceptor) [inline routes/rides.mjs]: Useless array

File: routes/rides.mjs [hunk 6]
```diff
@@ -169,16 +171,14 @@ const router = generateCRUD(Ride, {
...
     async main(ctx) {
       // @todo: Add right on max
       const { offset, limit } = ctx.parseRangePagination(Ride, { max: 1000 });
-      const start = new Date(ctx.query.filters.start);
-      const end = new Date(ctx.query.filters.end);
 
-      const total = await Ride.countDocumentsWithin(start, end, ctx.filters);
-      const data = await Ride.findWithin(start, end, ctx.filters).skip(offset).limit(limit).lean();
+      const total = await Ride.countDocuments(ctx.filters);
+      const data = await Ride.find(ctx.filters).skip(offset).limit(limit).lean();
```
Comment (Nainterceptor) [inline routes/rides.mjs]: Can you remove lean, please ? Will be useful in rights management at lower level.

File: middlewares/query-helper.mjs (hunk 1 above)
Comment (Scttpr) [reply middlewares/query-helper.mjs]: Fixed, not sure it's ideal, I think I need insights on this topic.

File: models/ride.mjs [hunk 7]
```diff
@@ -176,9 +176,45 @@ RideSchema.statics.castId = (v) => {
...
+  };
+
+  delete filter.start;
+  delete filter.end;
+
+
+  if (filter.current) {
+    let status;
+    if (filter.current === 'false') {
+      status = { status: 'delivered' };
```
Comment (Nainterceptor) [inline models/ride.mjs]: Please use consts from models
```js
import { DELIVERED } from '/models/status.mjs'; // Put right path
//...
status = { status: DELIVERED };
```

File: models/ride.mjs [hunk 8]
```diff
@@ -176,9 +176,45 @@ RideSchema.statics.castId = (v) => {
...
+    } else {
+      status = { $nor: [{ status: 'delivered' }] };
```
Comment (Nainterceptor) [inline models/ride.mjs]: Please use consts from models
```js
import { DELIVERED } from '/models/status.mjs'; // Put right path
//...
status = { $nor: [{ status: DELIVERED }] };
```

**Pull request merged**